nautobot/dcim/filters.py

```diff
@@ -1167,16 +1211,72 @@ class InterfaceFilterSet(
         queryset=Interface.objects.all(),
         label="Parent interface (ID)",
     )
+    parent_interface = NaturalKeyOrPKMultipleChoiceFilter(
+        to_field_name="name",
+        queryset=Interface.objects.all(),
+        label="Parent interface (name or ID)",
+    )
     bridge_id = django_filters.ModelMultipleChoiceFilter(
         field_name="bridge",
         queryset=Interface.objects.all(),
         label="Bridge interface (ID)",
     )
+    bridge = NaturalKeyOrPKMultipleChoiceFilter(
+        to_field_name="name",
+        queryset=Interface.objects.all(),
+        label="Bridge interface (name or ID)",
+    )
     lag_id = django_filters.ModelMultipleChoiceFilter(
         field_name="lag",
         queryset=Interface.objects.filter(type=InterfaceTypeChoices.TYPE_LAG),
         label="LAG interface (ID)",
     )
+    lag = NaturalKeyOrPKMultipleChoiceFilter(
+        to_field_name="name",
+        queryset=Interface.objects.all(),
+        label="LAG interface (name or ID)",
+    )
+    untagged_vlan = NaturalKeyOrPKMultipleChoiceFilter(
+        to_field_name="vid",
+        queryset=VLAN.objects.all(),
+        label="Untagged VLAN (VID or ID)",
+    )
+    tagged_vlans = NaturalKeyOrPKMultipleChoiceFilter(
+        to_field_name="vid",
+        queryset=VLAN.objects.all(),
+        label="Tagged VLANs (VID or ID)",
+    )
+    has_tagged_vlans = RelatedMembershipBooleanFilter(
+        field_name="tagged_vlans",
+        label="Has tagged VLANs",
+    )
+    child_interfaces = NaturalKeyOrPKMultipleChoiceFilter(
+        to_field_name="name",
+        queryset=Interface.objects.all(),
+        label="Child interfaces",
+    )
+    has_child_interfaces = RelatedMembershipBooleanFilter(
+        field_name="child_interfaces",
+        label="Has child interfaces",
+    )
+    bridged_interfaces = NaturalKeyOrPKMultipleChoiceFilter(
+        to_field_name="name",
+        queryset=Interface.objects.all(),
+        label="Bridged interfaces",
+    )
+    has_bridged_interfaces = RelatedMembershipBooleanFilter(
+        field_name="bridged_interfaces",
+        label="Has bridged interfaces",
+    )
+    member_interfaces = NaturalKeyOrPKMultipleChoiceFilter(
+        to_field_name="name",
+        queryset=Interface.objects.all(),
+        label="Member interfaces",
+    )
+    has_member_interfaces = RelatedMembershipBooleanFilter(
+        field_name="member_interfaces",
+        label="Has member interfaces",
+    )
     mac_address = MultiValueMACAddressFilter()
     tag = TagFilter()
```

Comment:
Looks like we should also have `ip_addresses` and `has_ip_addresses` filters similar to those being added to VMInterfaceFilterSet in #1938 - maybe a future PR though?

Reply:
My thought was to get all of the generic relations in the next round. I'll be opening a spike to figure out how we want to filter on GenericForeignKeys.